### PR TITLE
[Merged by Bors] - Bump post to v0.5.1

### LINKED ISSUE #217: Adjust POST algorithm to the latest agreements
We need to adjust POST algorithm to the recent agreements:
* `K2=8000`
* `K3=50`

`K3` is the subset of `K2` that is used to validate the proof. We need to make sure that each node will always validate the same `K3` indices from given K2 indices. We should use Fisher-Yates shuffle.

Besides that we should increase the bits per label to `16` so also we need to change `N` to script to `8192`.

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Bumps post to the new proving and verification algorithm

Closes https://github.com/spacemeshos/pm/issues/217

## Changes
- K2 and K3 Proof of Work for proving part,
- Verifying K3 subset of proofs,
- Labels are 16B long, scrypt difficulty is increased 16x to compensate, 
- Removed `BitsPerLabel` from configuration as it is fixed to 128bits now.

## Test Plan
All existing test cases should pass.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,7 +36,7 @@ require (
 	github.com/spacemeshos/go-scale v1.1.4
 	github.com/spacemeshos/merkle-tree v0.2.1
 	github.com/spacemeshos/poet v0.6.4
-	github.com/spacemeshos/post v0.5.0
+	github.com/spacemeshos/post v0.5.1-0.20230321145139-91e4ec956445
```
Comment (poszu) [inline go.mod]: Remember to bump to a released version when available and run `go mod tidy`.

File: cmd/root.go [hunk 2]
```diff
@@ -193,19 +193,22 @@ func AddCommands(cmd *cobra.Command) {
...
+		cfg.POST.K1, "difficulty factor for finding a good label when generating a proof")
 	cmd.PersistentFlags().Uint32Var(&cfg.POST.K2, "post-k2",
-		cfg.POST.K2, "")
+		cfg.POST.K2, "number of labels to prove")
+	cmd.PersistentFlags().Uint32Var(&cfg.POST.K3, "post-k3",
+		cfg.POST.K3, "subset of labels to verify in a proof")
+	cmd.PersistentFlags().Uint64Var(&cfg.POST.K2PowDifficulty, "post-k2pow-difficulty",
+		cfg.POST.K2PowDifficulty, "difficulty of K2 proof of work")
+	cmd.PersistentFlags().Uint64Var(&cfg.POST.K3PowDifficulty, "post-k3pow-difficulty",
+		cfg.POST.K3PowDifficulty, "difficulty of K3 proof of work")
```
Comment (fasmat) [inline cmd/root.go:211]: 👍 like the descriptions

**Review by fasmat — APPROVED** [review]
LGTM

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**